### PR TITLE
refactor(core): Have one orchestration service per instance type

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**
webhook instances will not listen to either worker or event log messages on the Redis pub/sub channel

### CODE REVIEW

File: packages/cli/src/services/orchestration.handler.service.ts [hunk 1]
```diff
@@ -9,12 +9,17 @@ import {
 import { handleWorkerResponseMessage } from './orchestration/handleWorkerResponseMessage';
 import { handleCommandMessage } from './orchestration/handleCommandMessage';
 import { MessageEventBus } from '../eventbus/MessageEventBus/MessageEventBus';
+import config from '@/config';
 
 @Service()
 export class OrchestrationHandlerService {
```
Comment (netroy) [inline packages/cli/src/services/orchestration.handler.service.ts:15]: does it make sense to break this service into 3 separate services (one for each instance type) ? That might help a lot against maintaining all the `if` checks, and might also help against us forgetting to put in those checks in code at times.

Comment (flipswitchingmonkey) [reply packages/cli/src/services/orchestration.handler.service.ts:15]: good idea

**Review by krynble — COMMENTED** [review]
I'll run some tests before approving, but wanted to leave those comments. In general, code looks good. I'll send a final review soon.

File: packages/cli/src/services/orchestration/main/orchestration.main.service.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	async broadcastRestartEventbusAfterDestinationUpdate() {
+		if (!this.sanityCheck()) return;
+		await this.redisPublisher.publishToCommandChannel({
+			command: 'restartEventBus',
+		});
+	}
+
+	async broadcastReloadExternalSecretsProviders() {
```
Comment (krynble) [inline packages/cli/src/services/orchestration/main/orchestration.main.service.ts:32]: Not a big deal, but when setting up external secrets for the first time, when I click save n8n under the hood does:
- save
- enable

This basically triggers this twice. Would be good if we could prevent this but not a deal breaker if too complciated. Maybe a debounce?

**Review by krynble — DISMISSED** [review]
Tested and changes are working as expected. Thanks for addressing this. Left a last minor comment.

**Review by krynble — DISMISSED** [review]

File: packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/services/orchestration/main/handleCommandMessageMain.ts:1]: this is in preparation for having multiple main instances, which would also need to be notified if one of them changes something in the settings

File: packages/cli/src/services/orchestration/webhook/handleCommandMessageWebhook.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { handleCommandMessageMain } from '../main/handleCommandMessageMain';
+
+export async function handleCommandMessageWebhook(messageString: string) {
+	// currently webhooks handle commands the same way as the main instance
```
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/services/orchestration/webhook/handleCommandMessageWebhook.ts:4]: see above

**Review by krynble — DISMISSED** [review]

File: packages/cli/src/services/orchestration/main/orchestration.main.service.ts (hunk 2 above)
Comment (flipswitchingmonkey) [reply packages/cli/src/services/orchestration/main/orchestration.main.service.ts:32]: added debouncing on the handler level

**Review by krynble — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with [`n8n@1.11.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.11.0)

**Pull request merged**